packages/abstractions/src/apiClientBuilder.ts

```diff
@@ -9,53 +9,58 @@ import { BackingStoreParseNodeFactory, BackingStoreSerializationWriterProxyFacto
 
 /**
  * Registers the default serializer to the registry.
+ * @param serializationWriterFactoryRegistry The registry to which the default serializer will be registered.
  * @param type the class of the factory to be registered.
  */
-export function registerDefaultSerializer(type: new () => SerializationWriterFactory): void {
+export function registerDefaultSerializer(serializationWriterFactoryRegistry: SerializationWriterFactoryRegistry, type: new () => SerializationWriterFactory): void {
```

Comment:
those should probably be instance methods now that we require an instance to be passed

Reply:
Had to revert to functions since they will require [cyclic dependencies](https://github.com/microsoft/kiota-typescript/actions/runs/13830232414/job/38692659039?pr=1634) that are [problematic](https://stackoverflow.com/questions/43176006/typeerror-class-extends-value-undefined-is-not-a-function-or-null)

Reply:
Moving `enableBackingStoreForParseNodeRegistry` to the parseNodeFactoryRegistry will introduce an import dependecy on `BackingStoreParseNodeFactory` which is extends `ParseNodeProxyFactory -> ParseNodeFactory` , since the registry also extends `ParseNodeFactory`, the issues is resulting from 'importing a file that is alsp being extended`

Reply:
right, thanks for articulating this!